src/lib/commands/create.ts

```diff
@@ -0,0 +1,201 @@
+const buildFileCreateListener = (
+  extension: string,
+  folderPath: string,
+  doAfter: Function,
+  cleanupCallback: Function
+) => {
+  return async (e: FileCreateEvent) => {
+    if (
+      e.files?.length === 1 &&
+      new RegExp(`${folderPath}/.*`).test(e.files[0].fsPath)
+    ) {
+      cleanupCallback();
+      const uniqueFilePath = getUniquePathName(e.files[0].fsPath, extension);
+      if (e.files[0].fsPath !== uniqueFilePath) {
+        const edit = new WorkspaceEdit();
+        edit.renameFile(e.files[0], Uri.file(uniqueFilePath));
+        await workspace.applyEdit(edit);
+      }
+      await doAfter(uniqueFilePath);
+    }
+  };
+};
+
+// doAfter receives (filepath: string)
+const onClickCreateFile = (extension: string, doAfter: Function) => {
+  return (clickContext: any) => {
+    if (clickContext.scheme === 'file') {
+      const createFileSubscription = workspace.onDidCreateFiles(
+        buildFileCreateListener(extension, clickContext.fsPath, doAfter, () =>
+          createFileSubscription.dispose()
+        )
+      );
+      commands.executeCommand('explorer.newFile');
+    }
+  };
+};
+const onClickCreateFolder = (extension: string, doAfter: Function) => {
+  return (clickContext: any) => {
+    if (clickContext.scheme === 'file') {
+      const createFileSubscription = workspace.onDidCreateFiles(
+        buildFileCreateListener(extension, clickContext.fsPath, doAfter, () =>
+          createFileSubscription.dispose()
+        )
+      );
+      commands.executeCommand('explorer.newFolder');
+    }
+  };
+};
+
+export const registerCommands = (context: ExtensionContext) => {
+  context.subscriptions.push(
+    commands.registerCommand(
+      COMMANDS.CREATE.SECTION,
+      onClickCreateFile('html', async (uniqueFilePath: string) => {
+        await createTemplate(
+          path.basename(uniqueFilePath),
+          path.dirname(uniqueFilePath),
+          TEMPLATE_NAMES.SECTION,
+          { allowExisting: true }
+        );
+        trackEvent(TRACKED_EVENTS.CREATE.SECTION);
+      })
+    )
+  );
+  context.subscriptions.push(
+    commands.registerCommand(
+      COMMANDS.CREATE.TEMPLATE,
+      onClickCreateFile('html', async (uniqueFilePath: string) => {
+        await createTemplate(
+          path.basename(uniqueFilePath),
+          path.dirname(uniqueFilePath),
+          TEMPLATE_NAMES.TEMPLATE,
+          { allowExisting: true }
+        );
+        trackEvent(TRACKED_EVENTS.CREATE.TEMPLATE);
+      })
+    )
+  );
+  context.subscriptions.push(
+    commands.registerCommand(
+      COMMANDS.CREATE.PARTIAL,
+      onClickCreateFile('html', async (uniqueFilePath: string) => {
+        await createTemplate(
+          path.basename(uniqueFilePath),
+          path.dirname(uniqueFilePath),
+          TEMPLATE_NAMES.PARTIAL,
+          { allowExisting: true }
+        );
+        trackEvent(TRACKED_EVENTS.CREATE.PARTIAL);
```

Comment:
I think `trackEvent` is async

Reply:
It is async, but since it doesn't interfere with any logic and we don't do anything with the result isn't it OK to omit the `await`?